### PR TITLE
src: export some methods and packages to allow for registering prebuilt docs

### PR DESCRIPTION
**Pull request by vavrusa**
Hi! I'm not sure if this is something you'd consider merging, but hopefully it's not too intrusive. The use case I have here is to support running just the docs.rs front-end and back-end without the rustwide builder. The crate docs are built in a separate process and uploaded to another service that calls these interfaces to register the package into the s3 and db, hence I need these interfaces visible.

### CODE REVIEW

File: src/utils/cargo_metadata.rs [hunk 1]
```diff
@@ -84,10 +83,26 @@ impl Package {
         self.library_target()
             .map(|target| self.normalize_package_name(&target.name))
     }
+
+    /// Deserialize metadata from lines.
+    pub fn from_lines<'a, T: AsRef<str>>(mut iter: impl Iterator<Item = T> + 'a) -> Result<Self> {
+        let metadata = if let (Some(serialized), None) = (iter.next(), iter.next()) {
```
Comment (jyn514) [inline src/utils/cargo_metadata.rs:89]: This seems unnecessarily generic - could you have it take `&[&str]` instead?

**Review by jyn514 — CHANGES_REQUESTED** [review]
These changes seem pretty reasonable to me - @rust-lang/docs-rs do you have opinions on merging this? I don't think there will be much ongoing maintenance necessary.

File: src/utils/cargo_metadata.rs (hunk 1 above)
Comment (vavrusa) [reply src/utils/cargo_metadata.rs:89]: Sure, I use it with [std::str::Lines](https://doc.rust-lang.org/std/str/struct.Lines.html) but I can collect it beforehand if you prefer that.

Comment (jyn514) [reply src/utils/cargo_metadata.rs:89]: Ah ok, that's fine then - I thought it was a slice to start.

**Comment by syphar** [comment]
@vavrusa I see you are still pushing to this branch. 

Could you ping me when this PR is ready for review?

**Comment by vavrusa** [comment]
@syphar sure thing. I'm trying to rebase, and the current version seems to require that the builds are also checked in the database, so I exported the necessary functions for those as well. I'll let you know when it works again, thanks!

**Comment by jyn514** [comment]
I'm going to close this for now, feel free to reopen if you get it working.

**Pull request closed without merging**